opentelemetry-sdk/setup.cfg

```diff
@@ -45,6 +45,7 @@ install_requires =
     opentelemetry-api == 1.5.0
     opentelemetry-semantic-conventions == 0.24b0
     opentelemetry-instrumentation == 0.24b0
+    typing-extensions ~= 3.10
```

Comment:
This allows us to use things like `typing_extensions.Protocol` to easily define factory types (callable). The end-result being a more helpful IDE and better static checks for end users.